v-next/hardhat/src/internal/builtin-plugins/coverage/hook-handlers/hre.ts

```diff
@@ -0,0 +1,44 @@
+import type { CoverageManager } from "../../../../types/coverage.js";
+import type { HardhatRuntimeEnvironmentHooks } from "../../../../types/hooks.js";
+
+import { getEmptyTmpDir } from "@nomicfoundation/hardhat-utils/fs";
+
+export class LazyCoverageManager implements CoverageManager {
+  #coverageManager: CoverageManager | undefined;
+
+  constructor() {
+    this.#coverageManager = undefined;
+  }
+
+  public async save(): Promise<void> {
+    const coverageManager = await this.#getCoverageManager();
+    return coverageManager.save();
+  }
+
+  async #getCoverageManager(): Promise<CoverageManager> {
+    if (this.#coverageManager === undefined) {
+      const { getOrCreateCoverageManager } = await import(
+        "../coverage-manager.js"
+      );
+      this.#coverageManager = getOrCreateCoverageManager();
+    }
+
+    return this.#coverageManager;
+  }
+}
+
+export default async (): Promise<Partial<HardhatRuntimeEnvironmentHooks>> => {
+  const handlers: Partial<HardhatRuntimeEnvironmentHooks> = {
+    created: async (_context, hre): Promise<void> => {
+      if (process.env.HARDHAT_COVERAGE_PATH === undefined) {
+        // NOTE: Saving the environment variable so that any subprocesses that
+        // inherit the env will operate within the same coverage path
+        process.env.HARDHAT_COVERAGE_PATH =
+          await getEmptyTmpDir("hardhat-coverage");
+      }
```

Comment:
This feels really hacky, too. As mentioned in https://github.com/NomicFoundation/hardhat/pull/6595/files#r2054463033, if this was a well-known directory, then we could maybe have an extra `clear` method on the coverage manager instead. Then the expected flow in the test tasks would be:
1. `clear` before any tests are run
2. `save` inside the test workers once the tests are complete
3. `load` after all the tests are done

Reply:
I explored this proposed API with additional `clear` method in https://github.com/NomicFoundation/hardhat/compare/coverage-6581...coverage-6581-alt and, in my opinion, clearing the coverage hits dir explicitly from the test task seems quite alright. Let me know what you think.